### PR TITLE
Don't allow PluginLoadCommand to add invalid plugins

### LINKED ISSUE #14870: PluginLoadCommand load should check if Plugin exists and CamelCase is correct
This is a (multiple allowed):

* [ ] bug
* [x] enhancement
* [ ] feature-discussion (RFC)

* CakePHP Version: 8f673ad
* Platform and Target: 

### What you did
`bin/cake plugin load NOTEXISTINGplugin
`
### What happened

> $this->addPlugin('NOTEXISTINGplugin');

gets added to Application.php, even if the plugin does not exists, breaking everything as the plugin could not be loaded. 

### What you expected to happen
Throw an exception/print an error if the user try's to load a plugin that does not exists or is spelled wrong.

### PR DESCRIPTION
**Pull request by markstory**
Catch mistakes earlier on as users can easily make typos when adding plugins to their applications.

Fixes #14870

### CODE REVIEW

File: src/Command/PluginLoadCommand.php [hunk 1]
```diff
@@ -69,6 +71,15 @@ public function execute(Arguments $args, ConsoleIo $io): ?int
             return static::CODE_ERROR;
         }
 
+        try {
+            Plugin::getCollection()->findPath($plugin);
+        } catch (MissingPluginException $e) {
+            $this->io->err('Could not find plugin <info>NopeNotThere</info>');
```
Comment (ADmad) [inline src/Command/PluginLoadCommand.php:77]: Need to use `$plugin`  instead of `NopeNotThere` in error message :slightly_smiling_face:.

Comment (YveIce) [reply src/Command/PluginLoadCommand.php:77]:
```suggestion
            $this->io->err($e->getMessage());
```

File: tests/TestCase/Command/PluginLoadCommandTest.php [hunk 2]
```diff
@@ -77,16 +77,31 @@ public function testHelp()
...
+    /**
+     * Test loading an unknown plugin
+     *
+     * @return void
+     */
+    public function testLoadUnknownPlugin()
+    {
+        $this->exec('plugin load NopeNotThere');
+        $this->assertExitCode(Command::CODE_ERROR);
+        $this->assertErrorContains('Could not find plugin <info>NopeNotThere</info>');
```
Comment (YveIce) [inline tests/TestCase/Command/PluginLoadCommandTest.php:102]:
```suggestion
        $this->assertErrorContains('Plugin NopeNotThere could not be found.');
```

**Review by YveIce — CHANGES_REQUESTED** [review]
I suggest to life with loosing the <info> tag, but using the default messageTemplate declared in /src/Command/MissingPluginException.php instead.

File: src/Command/PluginLoadCommand.php (hunk 1 above)
Comment (markstory) [reply src/Command/PluginLoadCommand.php:77]: I was pretty tired when I tried to do this and it shows. Sorry about that.

**Comment by dereuromark** [comment]
Based on the list of plugins inside the vendors plugin file, we could even output a list of possible plugins to chose from.

**Pull request merged**